posthog/api/capture.py

```diff
@@ -205,6 +205,10 @@ def get_event(request):
         if not distinct_id:
             continue
 
+        payload_uuid = event.get("uuid", None)
+        if payload_uuid and UUIDT.is_valid_uuid(payload_uuid):
+            event_uuid = UUIDT(uuid_str=payload_uuid)
```

Comment:
Let's respond with `invalid payload` instead if the uuid is defined but isn't valid as us then overriding it with a random one would be unexpected and difficult to debug in the future.

Reply:
Hmm this is a good point, but it's problematic here after the refactor.

People send us events in a batch, and we don't have an amazing story yet for how to handle invalid payloads within the batch. Sure we could respond with something like 'ingested these and not these' but that needs some careful consideration. it also would prevent us from improving latency down the line by dumping events into another Kafka queue immediately as we get them.

Currently invalid payloads get ignored, and in this case it's better to default to a random ID than it is to ignore imo.

Maybe a thing to do here is add metrics. Something for the future would also be a dashboard with an overview of ingestion.

Reply:
Anchor links seem broken but check out "Invalid events" here: https://posthog.com/docs/api/post-only-endpoints

Reply:
maybe we can add a log then (Sentry), so we at least know how often this is happening (so we know to prioritise)?

Reply:
Will add a statsd metric